management/src/commands/update-identity-from-user.ts

```diff
@@ -66,4 +75,41 @@ async function updateIdentityFromUser(
   });
 }
 
+async function updateIdentityFromUser(
+  managementCredentials: ManagementCredentials,
+  userId: string,
+  validationCode: string,
+  eventIds: string[],
+  identityValue: string,
+  identityToUpdateId: string,
+  maxRetry = 2,
+): Promise<void> {
+  return await updateIdentity(
+    managementCredentials,
+    userId,
+    validationCode,
+    eventIds,
+    identityValue,
+    identityToUpdateId,
+  ).catch(async (error) => {
+    if (error.statusCode >= 500 || error instanceof ConnectUnreachableError) {
+      if (RETRIES_COUNT < maxRetry) {
+        RETRIES_COUNT = RETRIES_COUNT + 1;
+        await delay();
+        return updateIdentityFromUser(
+          managementCredentials,
+          userId,
+          validationCode,
+          eventIds,
+          identityValue,
+          identityToUpdateId,
+          maxRetry,
+        );
+      }
+      RETRIES_COUNT = 0;
+    }
```

Comment:
Now that I think about it, I'm a bit scared of the global variable in this case. You only reset it in a specific flow, and I'm pretty sure that with this implementation, issues will be introduced because global variables are never disposed of by the garbage collector.

What about having your retry count as an initialised argument for `updateIdentityFromUser` ? Something along the lines of
```ts
async function updateIdentityFromUser(
  [...],
  retryCount = 0,
): Promise<void> {
  return await updateIdentity(
   [...],
  ).catch(async (error) => {
    if (error.statusCode >= 500 || error instanceof ConnectUnreachableError) {
      if (retriesCount < maxRetry) {
        await delay();
        return updateIdentityFromUser(
          [...],
          retriesCount + 1,
        );
      }
    }
    throw error;
  });
}
```
That way, the retry counter variable is disposed when function processing is done.
What do you think?

Reply:
To make this work, let's add `retryCount + 1` as an argument to the `delay` function but I agree with the comment

Reply:
I agree with that

Reply:
I'm a bit afraid of letting the retry counter "accessible" to the end user by making it an argument to `updateIdentityFromUser`. Imo this value shouldn't be editable, since it is meant to start at 0 and register each passage inside the function. "Never trust your users". Am I a little bit paranoid on this one ? 😅

Reply:
> "Never trust your users". Am I a little bit paranoid on this one ? 😅

No you're quite right in that, but that can be managed by exposing a function where you don't have this argument and using internally another function with it (that the exposed function call)

Reply:
By that, you mean adding a 3rd wrapper function above `updateIdentityFromUser` that will be called from the outside, without this argument?

Reply:
yes